vrs/marten-construction-script.py

```diff
@@ -0,0 +1,364 @@
+    # NOTE: would like to incorporate Multiallelic and Min-Rep into this dictionary (and above)
+    # BUT since they are generated very differently, it's difficult to work out.
+    desired_counts = {
+        "random": 50000,
+        "indel": 10000,
+        "long_ref": 10000,
+        "long_var": 10000,
+        "on_x": 9000,
+        "on_y": 1000,
+    }
+
+    # Reset the desired counts if the user chooses to change them - Multiallelic stored at [-2], Min-Rep at [-1]
+    counts_list = []
+    if args.custom:
+        counts_list = []
+        if args.custom:
+            argument_list = args.counts.split(",")
+            for argo in argument_list:
+                counts_list.append(argo)
+
+        for obj in enumerate(desired_counts.keys()):
+            desired_counts[obj[1]] = int(counts_list[obj[0]])
+
+    logger.info(f"Counts as: {desired_counts}")
+
+    # Annotate the Table with the status of Allele we are looking at
+    ht_whole = ht_whole.annotate(
+        indel=hl.is_indel(ht_whole.alleles[0], ht_whole.alleles[1]),
+        long_ref=hl.len(ht_whole.alleles[0]) > 5,
+        long_var=hl.len(ht_whole.alleles[1]) > 5,
+        on_x=(ht_whole.locus.contig == "chrX"),
+        on_y=(ht_whole.locus.contig == "chrY"),
+        random=(True),
+    )
+    logger.info("Annotation finished")
+
+    # Create Table with Reference, Indel, Long Reference, Long Variant, and Sex Chromosome variants
+    hts = []
+    for variant_type in freq_dict:
+        variant_freq = freq_dict[variant_type]
+        desired_count = desired_counts[variant_type]
+        sampling_freq = desired_count / (whole_count * variant_freq)
+
+        ht_var = ht_whole.filter(ht_whole[variant_type])
+        ht_var = ht_var.sample(sampling_freq)
+        ht_var = ht_var.checkpoint(
+            f"{args.tmp_dir}/{variant_type}_temp.ht",
+            _read_if_exists=args.read_if_exists,
+            overwrite=args.overwrite,
+        )
+        hts.append(ht_var)
+        logger.info(f"Finished creating table for %s", variant_type)
+
+    ht_union = hts[0].union(*hts[1:])
+
+    logger.info(
+        f"Total size of random, indel, long ref, long var, sex chr combined Table as: %i",
+        ht_union.count(),
+    )
+    ht_union = ht_union.checkpoint(
+        f"{args.tmp_dir}/union_path_temp.ht",
+        _read_if_exists=args.read_if_exists,
+        overwrite=args.overwrite,
+    )
+
+    # Construct a Table of only multiallelic sites
+
+    # For 3 random partitions, only take the highly multiallelic sites (>5 variants per loci)
+    # NOTE: Construct Table only by partitions, and each partition has ~3200 multiallelic variants
+    # 3 partitions to yield 9600 multiallelic variants (default, closest to 10,000)
+    multiallelic_partitions = 3
+
+    if args.custom:
+        multiallelic_partitions = int(counts_list[-2]) // 3200
```

Comment:
this 3200 should be a variable (rather than hardcoding here)

Reply:
won't be a problem if I introduce args.n_multi_partitions as an argument , but yes I will change